src/reformatters/contrib/noaa/ndvi_cdr/ndvi_cdr/analysis/region_job.py

```diff
@@ -0,0 +1,223 @@
+        """Download the file for the given coordinate and return the local path."""
+        url = coord.get_url()
+        # We set a special value if, when generating the source file coords, we did not find
+        # an available file.
+        if url == "no-url":
+            raise FileNotFoundError(f"No file found for {coord.time}")
+        return http_download_to_disk(url, self.dataset_id)
+
+    def read_data(
+        self,
+        coord: NoaaNdviCdrAnalysisSourceFileCoord,
+        data_var: NoaaNdviCdrDataVar,
+    ) -> ArrayFloat32 | ArrayInt16:
+        """Read and return an array of data for the given variable and source file coordinate."""
+        if data_var.name == "ndvi_usable":
+            return self._read_usable_ndvi(coord, data_var)
+        else:
+            return self._read_netcdf_data(coord, data_var)
+
+    def _read_netcdf_data(
+        self, coord: NoaaNdviCdrAnalysisSourceFileCoord, data_var: NoaaNdviCdrDataVar
+    ) -> ArrayFloat32 | ArrayInt16:
+        """Read data from NetCDF file."""
+        out_dtype = data_var.encoding.dtype
+        encoding_fill_value = data_var.encoding.fill_value
+
+        var_name = data_var.internal_attrs.netcdf_var_name
+        netcdf_fill_value = data_var.internal_attrs.fill_value
+        scale_factor = data_var.internal_attrs.scale_factor
+        add_offset = data_var.internal_attrs.add_offset
+
+        netcdf_path = f"netcdf:{coord.downloaded_path}:{var_name}"
+
+        with rasterio.open(netcdf_path) as reader:
+            masked_result = reader.read(1, out_dtype=out_dtype, masked=True)
+            result = masked_result.filled(netcdf_fill_value)
+
+            assert result.shape == (3600, 7200)
+            assert result.dtype == np.dtype(out_dtype)
+
+            # Set invalid values to NaN before scaling (for float data)
+            if var_name != QA_NETCDF_VAR_NAME:
+                result[result == netcdf_fill_value] = encoding_fill_value
+
+                assert scale_factor is not None
+                assert add_offset is not None
+                result *= scale_factor
+                result += add_offset
+
+                assert result.dtype == np.float32
+                return cast(ArrayFloat32, result)
+            else:
+                assert result.dtype == np.int16
+                return cast(ArrayInt16, result)
+
+    def _read_usable_ndvi(
+        self,
+        coord: NoaaNdviCdrAnalysisSourceFileCoord,
+        data_var: NoaaNdviCdrDataVar,
+    ) -> ArrayFloat32:
+        """Read NDVI data and apply quality filtering."""
+        ndvi_data = self._read_netcdf_data(coord, data_var)
+        qa_data = self._read_netcdf_data(coord, QA_DATA_VAR)
+
+        assert qa_data.dtype == np.int16
+        qa_data = cast(Array2D[np.int16], qa_data)
+
+        timestamp = pd.Timestamp(coord.time)
+
+        # Apply quality filtering based on timestamp
+        if timestamp < VIIRS_START_DATE:
```

Comment:
```suggestion
        # Apply quality filtering based on timestamp
        if coord.time < VIIRS_START_DATE:
```
was this giving you grief? our `Timestamp` type is already a pd.Timestamp, just annotated to play nicer with pydantic

Reply:
oh no, just overlooking what the type was, fixed